src/world/Script/NativeScriptApi.cpp

```diff
@@ -62,7 +62,7 @@ namespace Sapphire::ScriptAPI
   {
   }
 
-  void StatusEffectScript::onPlayerDamaged( Entity::Chara& actor )
+  void StatusEffectScript::onPlayerHit( Entity::CharaPtr& attackTarget, Sapphire::StatusEffect::StatusEffect& effect )
```

Comment:
i assume the intention here would've been entity::chara& instead of entity::charaptr&? passing a reference to the pointer is a bit moot (and scripts shouldn't ever get ownership either)